src/js/tools.js

```diff
@@ -37,14 +37,14 @@ function showFilters(element) {
 
 
 var closeplaylist;
-function overlayclickclose() {
+export function overlayclickclose() {
     if (closeplaylist) {
         $("#playlistdialog").dialog("close");
     }
     closeplaylist = 1;
 }
 
-function showPlaylistDialog(e, item_type, item_ids) {
+export function showPlaylistDialog(e, item_type, item_ids) {
     $("#playlistdialog").dialog("close");
 
     var parent = this;
```

Comment:
@mitchray tracked it back to here.
`TypeError: a is undefined`

could this be an issue where the dialog isn't actually there yet?

the div #playlistdialog doesn't exist and gets created by the function.

If i change this to the event (e), the dialog loads.

```
var parent = e;
```

is that wrong or incorrect somehow?

Reply:
set with bf69332923df75af25b38940f74e435e1a5b0e43 seems okay but just in case it's wrong

Reply:
Ah yeah the meaning of ```this``` changes now and no longer refers to the ```window``` object, I'll have to look out for that and update as I go.

_Technically_ that change works as its just building up some properties for an object to instantiate the dialog, and ```e``` happens to be an object; fine for the time being but I'll touch it up when I make the next pass